api/zulip/__init__.py

```diff
@@ -188,6 +188,21 @@ def __init__(self, email=None, api_key=None, config_file=None,
 
         if config_file is None:
             config_file = get_default_config_filename()
+
+        # Fill values from Environment Variables if not available in Constructor
+        if api_key is None:
+            api_key = os.environ.get("ZULIP_API_KEY")
+        if email is None:
+            email = os.environ.get("ZULIP_EMAIL")
+        if site is None:
+            site = get_env_variable("ZULIP_SITE")
```

Comment:
You forgot to change this one, which suggests you didn't test your work locally. 

Can you please verify that all of these options work?  E.g. I'd make a config file with invalid values for all of these and them confirm by passing the environment variables that you can make it all work.